demisto_sdk/commands/content_graph/objects/pack.py

```diff
@@ -603,3 +603,368 @@ def to_nodes(self) -> Nodes:
+                "contentDisplays": content_displays,
+                "dependencies": self.enhance_dependencies(marketplace),
+                "supportDetails": support_details,
+            }
+        )
+
+        return _metadata
+
+    def get_content_items_and_displays_metadata(
+        self, marketplace: MarketplaceVersions
+    ) -> Tuple[Dict, Dict]:
+        """
+        Gets the pack content items and display names to add into the pack's metadata dictionary.
+        For each content item the function generates its `summary` and calculating the from/to version
+        on whether to add this item to the content items list.
+
+        Args:
+            marketplace (MarketplaceVersions): The marketplace to which the pack should belong to.
+
+        Returns:
+            Tuple[Dict, Dict]: The content items and display names dictionaries to add to the pack metadata.
+        """
+        content_items: dict = {}
+        content_displays: dict = {}
+        for content_item in self.content_items:
+
+            if content_item.content_type == ContentType.TEST_PLAYBOOK:
+                logger.debug(
+                    f"Skip loading the {content_item.name} test playbook into metadata.json"
+                )
+                continue
+
+            add_item_to_metadata_list(
+                collected_content_items=content_items,
+                content_item=content_item,
+                marketplace=marketplace,
+            )
+
+            content_displays[
+                content_item.content_type.metadata_name
+            ] = content_item.content_type.metadata_display_name
+
+        content_displays = {
+            content_type: content_type_display
+            if (content_items[content_type] and len(content_items[content_type]) == 1)
+            else f"{content_type_display}s"
+            for content_type, content_type_display in content_displays.items()
+        }
+
+        return content_items, content_displays
+
+    def enhance_dependencies(self, marketplace):
+        """
+        Gathers the first level pack's dependencies details to a list to add to the pack's metadata.
+        For each dependency it adds the following pack's properties:
+        `mandatory`, `minVersion`, `author`, `name`, `certification`
+
+        Args:
+            marketplace (MarketplaceVersions): The marketplace to which the pack should belong to.
+
+        Returns:
+            dict: The dependencies of the pack.
+        """
+        return {
+            r.content_item_to.object_id: {
+                "mandatory": r.mandatorily,
+                "minVersion": r.content_item_to.current_version,  # type:ignore[attr-defined]
+                "author": r.content_item_to.author  # type:ignore[attr-defined]
+                if marketplace == MarketplaceVersions.XSOAR
+                else r.content_item_to.author.replace(  # type:ignore[attr-defined]
+                    "XSOAR", "XSIAM"
+                ),
```

Comment:
let's use the same method I suggested above

Reply:
done